### PR TITLE
Town Warps/Town+Nation Banners/EnhancedChat/Config Edits/Command Edits

### PR DESCRIPTION
**Pull request by Wowserman**

### CODE REVIEW

File: src/com/palmergames/bukkit/towny/TownySettings.java [hunk 1]
```diff
@@ -1805,11 +1811,6 @@ public static boolean isTownRespawningInOtherWorlds() {
 
 		return getBoolean(ConfigNodes.GTOWN_SETTINGS_TOWN_RESPAWN_SAME_WORLD_ONLY);
 	}
-	
-	public static int getMaxResidentsPerTown() {
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/TownySettings.java]: Probably dont want this removed.

File: src/com/palmergames/bukkit/towny/command/NationCommand.java [hunk 2]
```diff
@@ -65,7 +66,7 @@
 		nation_help.add(ChatTools.formatTitle("/nation"));
 		nation_help.add(ChatTools.formatCommand("", "/nation", "", TownySettings.getLangString("nation_help_1")));
 		nation_help.add(ChatTools.formatCommand("", "/nation", TownySettings.getLangString("nation_help_2"), TownySettings.getLangString("nation_help_3")));
-		nation_help.add(ChatTools.formatCommand("", "/nation", "list", TownySettings.getLangString("nation_help_4")));
+		nation_help.add(ChatTools.formatCommand("", "/nation", "list .. (nation)", TownySettings.getLangString("nation_help_4")));
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/command/NationCommand.java]: not sure why you have (nation) added after list, the command is /nation list

File: src/com/palmergames/bukkit/towny/command/NationCommand.java [hunk 3]
```diff
@@ -712,7 +726,7 @@ public static void nationAdd(Player player, Nation nation, List<Town> invited) {
 		for (Town town : invited)
 			try {				
 				if ((TownySettings.getNumResidentsJoinNation() > 0) && (town.getNumResidents() < TownySettings.getNumResidentsJoinNation())) {
-		        	TownyMessaging.sendErrorMsg(player, String.format(TownySettings.getLangString("msg_err_not_enough_residents_join_nation"), town.getName()));
+		        	TownyMessaging.sendErrorMsg(player, String.format(TownySettings.getLangString("msg_err_not_enough_residents_join_nation")));
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/command/NationCommand.java]: How come this is changed?

File: src/com/palmergames/bukkit/towny/command/NationCommand.java [hunk 4]
```diff
@@ -694,7 +708,7 @@ public void nationAdd(Player player, String[] names) {
 			nation = resident.getTown().getNation();
 			
 	        if ((TownySettings.getNumResidentsJoinNation() > 0) && (resident.getTown().getNumResidents() < TownySettings.getNumResidentsJoinNation())) {
-	        	TownyMessaging.sendErrorMsg(player, String.format(TownySettings.getLangString("msg_err_not_enough_residents_join_nation"), resident.getTown().getName()));
+	        	TownyMessaging.sendErrorMsg(player, String.format(TownySettings.getLangString("msg_err_not_enough_residents_join_nation")));
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/command/NationCommand.java]: How come this is changed?

File: src/com/palmergames/bukkit/towny/command/TownCommand.java [hunk 5]
```diff
@@ -124,10 +129,13 @@ private void parseTownCommand(Player player, String[] split) {
 
 			} else if (split[0].equalsIgnoreCase("list")) {
 
-				if (!TownyUniverse.getPermissionSource().testPermission(player, PermissionNodes.TOWNY_COMMAND_TOWN_LIST.getNode()))
+				if (!TownyUniverse.getPermissionSource().testPermission(player, PermissionNodes.TOWNY_COMMAND_NATION_LIST.getNode()))
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/command/TownCommand.java]: Wrong node being checked here

File: src/com/palmergames/bukkit/towny/command/TownCommand.java [hunk 6]
```diff
@@ -64,6 +67,7 @@
 		output.add(ChatTools.formatCommand("", "/town", "list", ""));
 		output.add(ChatTools.formatCommand("", "/town", "online", TownySettings.getLangString("town_help_10")));
 		output.add(ChatTools.formatCommand("", "/town", "leave", ""));
+		output.add(ChatTools.formatCommand("", "/town", "evict", ""));
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/command/TownCommand.java]: The evict command just sends a player who isn't a resident of your town, out of the town somewhere?

File: src/com/palmergames/bukkit/towny/command/TownCommand.java [hunk 7]
```diff
@@ -1933,9 +2073,6 @@ public static void townAddResidents(Object sender, Town town, List<Resident> inv
 					} else if (!TownyUniverse.getPermissionSource().has(BukkitTools.getPlayer(newMember.getName()), PermissionNodes.TOWNY_TOWN_RESIDENT.getNode())) {
 						TownyMessaging.sendErrorMsg(sender, String.format(TownySettings.getLangString("msg_not_allowed_join"), newMember.getName()));
 						invited.remove(newMember);
-					} else if (TownySettings.getMaxResidentsPerTown() > 0 && town.getResidents().size() >= TownySettings.getMaxResidentsPerTown()){
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/command/TownCommand.java]: Probably dont want this removed.

File: src/com/palmergames/bukkit/towny/listeners/TownyCustomListener.java [hunk 8]
```diff
@@ -60,7 +60,7 @@ public void onPlayerChangePlotEvent(PlayerChangePlotEvent event) {
 					player.sendMessage(msg);
 			}
 		} catch (NotRegisteredException e) {
-			// likely Citizens' NPC
+			e.printStackTrace();
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/listeners/TownyCustomListener.java]: This is left blanked out so that we dont see stack traces.

File: src/com/palmergames/bukkit/towny/permissions/PermissionNodes.java [hunk 9]
```diff
@@ -79,7 +81,13 @@
 			TOWNY_COMMAND_TOWN_SET_NAME("towny.command.town.set.name"),
 			TOWNY_COMMAND_TOWN_SET_TAG("towny.command.town.set.tag"),
 			TOWNY_COMMAND_TOWN_SET_JAIL("towny.command.town.set.jail"),
-		
+			TOWNY_COMMAND_TOWN_SET_WARP("towny.command.town.set.warp"), //TODO: Finish Implementation
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/permissions/PermissionNodes.java]: All of these TODOs, did they get done? Do the commands still need to have permission checks done on them?

File: src/com/palmergames/bukkit/towny/war/eventwar/WarTimerTask.java [hunk 10]
```diff
@@ -62,7 +62,7 @@ public void run() {
 						//TODO: Cache player coord & townblock
 
 						WorldCoord worldCoord = new WorldCoord(player.getWorld().getName(), Coord.parseCoord(player));
-						if (!warEvent.isWarZone(worldCoord))
+						if (!War.isWarZone(worldCoord))
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/war/eventwar/WarTimerTask.java]: Why was this changed?

File: src/com/palmergames/bukkit/towny/war/eventwar/WarTimerTask.java [hunk 11]
```diff
@@ -132,7 +132,7 @@ public static boolean isOnEdgeOfTown(TownBlock townBlock, WorldCoord worldCoord,
 			try {
 				TownBlock edgeTownBlock = worldCoord.getTownyWorld().getTownBlock(new Coord(worldCoord.getX() + offset[i][0], worldCoord.getZ() + offset[i][1]));
 				boolean sameTown = edgeTownBlock.getTown() == townBlock.getTown();
-				if (!sameTown || (sameTown && !warEvent.isWarZone(edgeTownBlock.getWorldCoord()))) {
+				if (!sameTown || (sameTown && !War.isWarZone(edgeTownBlock.getWorldCoord()))) {
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/war/eventwar/WarTimerTask.java]: Why was this changed?

File: src/english.yml [hunk 12]
```diff
@@ -1,5 +1,5 @@
 name: Towny
-version: 0.34
+version: 0.33
```
Comment (LlmDl) [inline src/english.yml]: All your language files are out of date. I don't think you started from the latest commits of mine. You will need to make sure that you have all of my commits before I can accept anything.

File: src/plugin.yml [hunk 13]
```diff
@@ -1,13 +1,14 @@
 name: Towny
 main: com.palmergames.bukkit.towny.Towny
-version: 0.91.4.12
-author: Shade, Modified by FuzzeWuzze. Forked by ElgarL. Forked by LlmDl.
+version: 0.91.4.10
```
Comment (LlmDl) [inline src/plugin.yml]: Again, you aren't running off the latest commits of mine.

File: src/plugin.yml [hunk 14]
```diff
@@ -1,13 +1,14 @@
...
+authors: [Shade, FuzzeWuzze, ElgarL, LlmDl, Wowserman]
 website: http://code.google.com/a/eclipselabs.org/p/towny/
 description: >
     Resident-Town-Nation heirarchy combined with a grid based
     protection system. Including a war event.
+    Created by Shade, Modified by FuzzeWuzze. Forked by ElgarL. Forked by LlmDl. Sweetened by Wowserman.
 
-softdepend: [MultiVerse,Multiverse-Core,XcraftGate,My Worlds,Citizens,PersonalWorlds,WormholeXTremeWorlds,Vault,MultiWorld]
+softdepend: [MultiVerse,Multiverse-Core,XcraftGate,My Worlds,Citizens,PersonalWorlds,WormholeXTremeWorlds,Vault,MultiWorld,EnhancedChat]
```
Comment (LlmDl) [inline src/plugin.yml]: The enhanced chat requires a separate plugin?

**Comment by LlmDl** [comment]
I've made a lot of comments. There are major conflicts in your PR since you've started with an out of date version of Towny.

Also not sure if I want to have to go through the menu/banner/warp stuff with this go around. I was hoping for just the enchanced TownyChat stuff and the underlying API stuff  in Towny that would be required for it to work.

File: .classpath [hunk 15]
```diff
@@ -20,3 +21,26 @@
 	<classpathentry kind="lib" path="/Users/peter/Downloads/craftbukkit-1.12.1.jar"/>
```
Comment (LlmDl) [inline .classpath]: The classpath has been altered to include your own directory as parent.

**Comment by Wowserman** [comment]
Thought this pull was closed. I merged it all.

File: src/ChangeLog.txt [hunk 16]
```diff
@@ -2526,11 +2526,11 @@ v0.91.4.8:
     - Possible fix for outpost spawns not being removed from the database when outposts are unclaimed.
     - Add more detail to messages shown in log when Towny enters safe mode. 
 v0.91.4.9:
-    - Remove leading space on the "(PVP)" chunk notification, to match the the no-pvp-message. Thanks for pointing this out Locknlol
```
Comment (Wowserman) [inline src/ChangeLog.txt]: Funky Merge, Removed and Re-Added lines. In the file it still exists.

File: src/ChangeLog.txt [hunk 17]
```diff
@@ -2526,11 +2526,11 @@ v0.91.4.8:
...
+    - Remove leading spaceon the "(PVP)" chunk notification, to match the the no-pvp-message. Thanks for pointing this out Locknlol
     - Fix for servers that do not allow wilderness full build/destroy permissions, where animals could not be killed in the wilderness with arrows.
     - Fix a typo in the CombatUtil, which clarifies which block's destroy permissions we are testing to determine whether animals can be killed.
     - Bump language files to version 0.33   
-    - New Config Option: town.max_claim_radius_value
+    - New Config Optin: town.max_claim_radius_value
```
Comment (Wowserman) [inline src/ChangeLog.txt]: Funky Merge, Removed and Re-Added lines. In the file it still exists. Again, not sure why this happened.

File: src/ChangeLog.txt [hunk 18]
```diff
@@ -2557,17 +2557,4 @@ v0.91.4.10:
     - Fix mayors being able to be made into Outlaws.
     - Fix an issue with plotManagementRevertSpeed on mysql databases.
     - Fix issue plaguing Mysql DB's in 0.91.4.9 where the console was spammed endlessly by revert speed default settings.
-    - Fix issue with outposts not unclaim their outpostspawns.
-v0.91.4.11:
-    - Fix 'Format specifier "%s"' error seen when attempting to add a town to you nation that doesn't have the config's required_number_residents_join_nation.
```
Comment (Wowserman) [inline src/ChangeLog.txt]: Funky Merge, Removed and Re-Added lines. In the file it still exists. Happened again.

File: src/com/palmergames/bukkit/config/ConfigNodes.java [hunk 19]
```diff
@@ -357,10 +357,14 @@
...
 			"  ############################################################",
 			""),
+	GTOWN_SETTINGS_MENU_LISTING(
+			"global_town_settings.menu_page-inator",
+			"true",
+			"# Should we Show the Fancy Town and Nation Lists (GUI)? Will use default text page-inator if false."),
 	GTOWN_SETTINGS_FRIENDLY_FIRE(
 			"global_town_settings.friendly_fire",
 			"true",
-			"# can residents/Allies harm other residents when in an area with pvp enabled? Other than an Arena plot."),
```
Comment (Wowserman) [inline src/com/palmergames/bukkit/config/ConfigNodes.java]: Funky Merge, Removed and Re-Added lines. In the file it still exists. Happened again.

File: src/com/palmergames/bukkit/config/ConfigNodes.java [hunk 20]
```diff
@@ -357,10 +357,14 @@
 			"  # +------------------------------------------------------+ #",
 			"  ############################################################",
 			""),
+	GTOWN_SETTINGS_MENU_LISTING(
```
Comment (Wowserman) [inline src/com/palmergames/bukkit/config/ConfigNodes.java]: Lets the fancy GUI thing be turned off.

File: .classpath [hunk 21]
```diff
@@ -1,21 +1,21 @@
...
-	<classpathentry kind="lib" path="../lib/bukkit.jar">
-		<attributes>
-			<attribute name="javadoc_location" value="http://jd.bukkit.org/apidocs/"/>
-		</attributes>
-	</classpathentry>
-	<classpathentry kind="lib" path="../lib/citizensapi-2.0-SNAPSHOT.jar"/>
-	<classpathentry kind="lib" path="../lib/Vault.jar"/>
-	<classpathentry kind="output" path="bin"/>
-</classpath>
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (LlmDl) [inline .classpath]: All these paths are messed up, missing the ../ , and we don't build using craftbukkit and we don't all have a /Users/peter/Downloads/ folder.

File: legacy/TownyHModFlatFileSource.java [hunk 22]
```diff
@@ -1,3 +1,487 @@
+<<<<<<< HEAD:src/com/palmergames/bukkit/towny/db/TownyHModFlatFileSource.java
```
Comment (LlmDl) [inline legacy/TownyHModFlatFileSource.java]: You don't need to add anything to the hmodflatfilesource, it is legacy and will probably be moved to the legacy folder soonish

File: src/ChangeLog.txt [hunk 23]
```diff
@@ -2557,6 +2557,9 @@ v0.91.4.10:
     - Fix mayors being able to be made into Outlaws.
     - Fix an issue with plotManagementRevertSpeed on mysql databases.
     - Fix issue plaguing Mysql DB's in 0.91.4.9 where the console was spammed endlessly by revert speed default settings.
+<<<<<<< HEAD
+    - Fix issue with outposts not unclaim their outpostspawns.
```
Comment (LlmDl) [inline src/ChangeLog.txt]: dont want this merge stuff here

File: src/ChangeLog.txt [hunk 24]
```diff
@@ -2695,4 +2698,5 @@ v0.92.0.2:
       - This event can be called by other region plugins, to cancel Towny claims inside of them. 
       - When the event is cancelled, Towny will stop short of charging for the townblock.
     - Language files bumped to 0.38.
-      - Italian.yml updated by Leomixer17.
\ No newline at end of file
+      - Italian.yml updated by Leomixer17.
+>>>>>>> upstream/master
```
Comment (LlmDl) [inline src/ChangeLog.txt]: or here

File: src/com/palmergames/bukkit/config/ConfigNodes.java (hunk 20 above)
Comment (LlmDl) [reply src/com/palmergames/bukkit/config/ConfigNodes.java]: Not sure if we want the fancy GUI just yet

File: src/com/palmergames/bukkit/towny/Towny.java [hunk 25]
```diff
@@ -127,14 +145,25 @@ public void onEnable() {
...
+		}
+		*/
 
 		registerEvents();
 
 		TownyLogger.log.info("=============================================================");
 		if (isError())
 			TownyLogger.log.info("[WARNING] - ***** SAFE MODE ***** " + version);
-		else
+		else {
```
Comment (LlmDl) [inline src/com/palmergames/bukkit/towny/Towny.java]: Can skip this { if the only line its there for is commented out.
